### PR TITLE
[Instrumentation.Runtime] Reword README.md

### PR DESCRIPTION
**Pull request by xiang17**
Fixes #.

## Changes

* Reword README.md on how to enable runtime instrumentation
* Remove the troubleshooting section

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [x] Design discussion issue https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/553#issuecomment-1201709502

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 1]
```diff
@@ -22,9 +22,11 @@ dotnet add package OpenTelemetry.Instrumentation.Runtime
 
 ### Step 2: Enable runtime instrumentation
 
-Runtime instrumentation should be enabled at application startup. This is
-typically done in the `ConfigureServices` of your `Startup` class. The example
-below enables this instrumentation by using an extension method on
+Runtime instrumentation should be enabled at application startup.
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:25]: you can keep this doc very simple by explaining the extension method and linking to the examples. Does not have to deal with asp.net core vs others here.

Comment (xiang17) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md:25]: I realized that the doc for [OpenTelemetry.Extensions.Hosting](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Extensions.Hosting) doesn't have any info on the `AddOpenTelemetryMetrics` extension, or how to configure metrics on ASP.NET Core applications.

Instead, I can only find it in the README of [Prometheus Exporter AspNetCore](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Prometheus.AspNetCore/README.md#step-2-configure-opentelemetry-meterprovider).

Comment (xiang17) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md:25]: Perhaps it's better to deal with the asp.net core vs others in the read me  of OpenTelemetry.Extensions.Hosting.

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 2]
```diff
@@ -60,6 +35,11 @@ using var meterProvider = Sdk.CreateMeterProviderBuilder()
 Refer to [Program.cs](../../examples/runtime-instrumentation/Program.cs) for a
 complete demo.
 
+Additionally, this examples sets up the OpenTelemetry Prometheus exporter, which
+requires adding the package
+[`OpenTelemetry.Exporter.Prometheus`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Prometheus/README.md)
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:40]: this is a dead link

File: src/OpenTelemetry.Instrumentation.Runtime/README.md (hunk 1 above)
Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md]: yes. Its not the concern of this runtime instrumentation that other components are not well documented.

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 3]
```diff
@@ -22,33 +22,8 @@ dotnet add package OpenTelemetry.Instrumentation.Runtime
...
-    services.AddOpenTelemetryMetrics((builder) => builder
-        .AddRuntimeInstrumentation()
-        .AddPrometheusExporter()
-    );
-}
-```
-
-Or configure directly:
+Runtime instrumentation should be enabled at application startup using the
+`AddRuntimeInstrumentation` extension:
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:26]:
```suggestion
`AddRuntimeInstrumentation` extension on `MeterProviderBuilder`:
```

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 4]
```diff
@@ -316,15 +296,6 @@ Relevant API:
   Occurs when an exception is thrown in managed code, before the runtime searches
   the call stack for an exception handler in the application domain.
 
-## Troubleshooting
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:319]: I'd suggest keeping the section.
We can add quick notes about.
if a metric is missing, review the doc to see if the metric is available in .NET version you are running.
also mention those metrics which are only available after GC has be completed atleast once?

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 5]
```diff
@@ -60,6 +35,11 @@ using var meterProvider = Sdk.CreateMeterProviderBuilder()
 Refer to [Program.cs](../../examples/runtime-instrumentation/Program.cs) for a
 complete demo.
 
+Additionally, this examples sets up the OpenTelemetry Prometheus exporter, which
+requires adding the package
+[`OpenTelemetry.Exporter.Prometheus`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Prometheus.AspNetCore/README.md)
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:40]: use httplistener one as this example seems to be just console app, and not asp.net core

Comment (xiang17) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md]: I updated the link and the text. However we don't have the `OpenTelemetry.Exporter.Prometheus.HttpListener` NuGet package yet. I'll probably keep the package name until we release a new package later.

Comment (xiang17) [reply src/OpenTelemetry.Instrumentation.Runtime/README.md]: https://www.nuget.org/packages?q=+OpenTelemetry.Exporter.Prometheus

File: src/OpenTelemetry.Instrumentation.Runtime/README.md [hunk 6]
```diff
@@ -318,12 +298,11 @@ Relevant API:
 
 ## Troubleshooting
 
-This component uses an
-[EventSource](https://docs.microsoft.com/dotnet/api/system.diagnostics.tracing.eventsource)
-with the name "OpenTelemetry-Instrumentation-Runtime" for its internal
-logging. Please refer to [SDK
-troubleshooting](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry#troubleshooting)
-for instructions on seeing these internal logs.
+If a metric is missing, review the doc to see if the metric is available in the
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Runtime/README.md:301]:
```suggestion
If a metric is missing, review the [list of metrics](#Metrics) to see if the metric is available in the
```

**Review by cijothomas — APPROVED** [review]

**Pull request merged**